plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/FieldSetterFactory.java

```diff
@@ -459,4 +473,107 @@ public void setField(Block block, int position)
             rowInspector.setStructFieldData(row, field, value);
         }
     }
+
+    protected Object getField(Type type, Block block, int position)
```

Comment:
This is copied from `io.trino.plugin.hive.util.HiveWriteUtils#getField`.
The functions can go out of sync.

I think that instead of copying and customizing for parquet and timestamp, it would be more maintainable to keep the single implementation in `io.trino.plugin.hive.util.HiveWriteUtils#getField` and pass timestamp `DateTimeZone` there. It would be UTC unless Parquet. I know it would not be as cleanly object-orientedly designed as your code here, but i still think it would be less err prone for the future, eg should we add support for TIME type.

The `io.trino.plugin.hive.util.HiveWriteUtils#getField` has two call sites today, so adding a mandatory arg (the zone) isn't a big deal.

Reply:
That was actually the first idea that I had, but I decided against it because we need a better solution to Parquet's unusual time zone behavior than just passing an extra parameter to everything in the code base.

That was what I was working towards in the commits now removed from this PR—I think I came within a few steps of a full solution for the duplicated type-translation logic (if I extracted some of that code into a separate class, it should cover everything except for partitioning on timestamps in Parquet).

I _could_ go back and redo this to use an extra time zone parameter, but I'd rather finish the work I started on [this branch](https://github.com/jirassimok/presto/commits/parquet-type-cleanup).

Reply:
Done.